src/rdkafka.h

```diff
@@ -9848,6 +9869,221 @@ RD_EXPORT void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+ * needs to be performed.
+ *
+ * @returns a new allocated ElectLeaders object.
+ *          Use rd_kafka_ElectLeaders_destroy() to free object when done.
+ */
+RD_EXPORT rd_kafka_ElectLeaders_t *
+rd_kafka_ElectLeaders_new(rd_kafka_ElectionType_t election_type,
+                          rd_kafka_topic_partition_list_t *partitions);
+
+/**
+ * @brief Destroy and free an ElectLeaders object previously created with
+ *        rd_kafka_ElectLeaders_new()
+ *
+ * @param elect_leaders The ElectLeaders object to be destroyed.
+ *
+ */
+RD_EXPORT void
+rd_kafka_ElectLeaders_destroy(rd_kafka_ElectLeaders_t *elect_leaders);
+
+/**
+ * @brief Elect Leaders for the provided Topic Partitions
+ *        according to the specified election type.
+ *
+ * @param rk Client instance.
+ * @param elect_leaders The elect leaders request containing
+ *          election type and partitions information.
+ * @param options Optional admin options, or NULL for defaults.
+ * @param rkqu Queue to emit result on.
+ *
+ * Supported admin options:
+ *  - rd_kafka_AdminOptions_set_operation_timeout() - default 60 seconds.
+ *    Controls how long the brokers will wait for records to be deleted.
+ *  - rd_kafka_AdminOptions_set_request_timeout() - default socket.timeout.ms.
+ *    Controls how long \c rdkafka will wait for the request to complete.
+ *
+ * @remark The result event type emitted on the supplied queue is of type
+ *         \c RD_KAFKA_EVENT_ELECTLEADERS_RESULT
+ */
+RD_EXPORT void rd_kafka_ElectLeaders(rd_kafka_t *rk,
+                                     rd_kafka_ElectLeaders_t *elect_leaders,
+                                     const rd_kafka_AdminOptions_t *options,
+                                     rd_kafka_queue_t *rkqu);
+
+/**
+ * @brief A struct consisting of Topic Partition result
+ *        including custom error message.
+ */
+typedef struct rd_kafka_topic_partition_result_s
+    rd_kafka_topic_partition_result_t;
+
+/**
+ * @brief  A struct representing result of elect leaders admin operation.
+ */
+typedef struct rd_kafka_ElectLeadersResult_s rd_kafka_ElectLeadersResult_t;
+
+/**
+ * @brief Get the topic name from the rd_kafka_topic_partition_result_t object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object.
+ *
+ * @returns the topic name from the rd_kafka_topic_partition_result_t object.
+ */
+RD_EXPORT const char *rd_kafka_topic_partition_result_topic(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Get the partition number from the rd_kafka_topic_partition_result_t
+ * object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object.
+ *
+ * @returns the partition number from the rd_kafka_topic_partition_result_t
+ * object.
+ */
+RD_EXPORT int32_t rd_kafka_topic_partition_result_partition(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Get the error code from the rd_kafka_topic_partition_result_t object.
```

Comment:
I would suggest not to use the struct name directly. Same at other places.

```suggestion
 * @brief Get the error code from the topic partition result.
```